### PR TITLE
Fixing three bugs in ElementInspector.js

### LINKED ISSUE #475: Change repeat end value drop down menu
Add a feature so you can not select an end repeat value that is only one segment away from the start

### LINKED ISSUE #501: Fix which functions are selectable
Add a feature where if the previous/initial sat value is denied you can't select decrease and if the previous/initial sat value is satisfied you can't select increase

### LINKED ISSUE #502: When an NBT link is selected set function type for both intentions to NBT and clear functions segments

### PR DESCRIPTION
**Pull request by i-m-foster**
Setting NB functions correctly, disabling drop down functions depending on the initial satisfaction value, and changing the repeat-end drop down menu so you cannot select an invalid end repeat value

### CODE REVIEW

**Comment by YesugenB** [comment]
this pr sucks

File: leaf-ui/rappid-extensions/ElementInspector.js [hunk 1]
```diff
@@ -399,30 +411,34 @@ var ElementInspector = Backbone.View.extend({
...
+        var minimumInt = string1.charCodeAt(0);
+        var nextChar = String.fromCharCode(begin.charCodeAt(0) + 2);
+
+        // Disable repeat end values from 'B' to one value after the repeat beginning value
+        // Because repeating segments must be two or more segments apart 
+        for (var i = (minimumInt); i < nextChar.charCodeAt(0); i++){
+            valOption = String.fromCharCode(i);
+            var disabledOpt = 'option[value=' + valOption + ']'
+            this.$(disabledOpt).prop('disabled', 'disabled');
+            // this.$("#repeat-begin").prop('disabled', false);
```
Comment (Domily11) [inline leaf-ui/rappid-extensions/ElementInspector.js]: Remove this?

**Comment by amgrubb** [comment]
- [x] After turning off NBN on the relationship the intention icon disappears, but when you click on them, they reappear and the side bar is disabled.

Screen Shot 2021-07-22 at 10 14 16 AM

**Comment by amgrubb** [comment]
- [x] To clarify intended behavior, you can have a repeating function over a single segment. In this case, you can have A-B as the repeat.

Screen Shot 2021-07-22 at 10 16 38 AM

File: leaf-ui/rappid-extensions/ElementInspector.js [hunk 2]
```diff
@@ -156,6 +155,24 @@ var ElementInspector = Backbone.View.extend({
...
+        // If the function is NB clear the functionSegList, set the function type to NB, and set the initial satisfaction value to none
+        if (!this.model.attr(".satvalue/value") && this.model.attr(".funcvalue/text") == "NB") {
+            if (this.intention.get('evolvingFunction') != null) {
+                this.intention.get('evolvingFunction').set('functionSegList', []);
+                this.intention.get('evolvingFunction').set('type', 'NB');
+                this.intention.getUserEvaluationBBM(0).set('assignedEvidencePair', '0000')
+            }
+        }
+
+        // If function used to be NB, reset the function type and initial satisfaction value
```
Comment (amgrubb) [inline leaf-ui/rappid-extensions/ElementInspector.js:167]: I appreciate these comments.

**Review by amgrubb — APPROVED** [review]
Bugs fixed, nicely done. `Squash and merge`!

**Review by YesugenB — APPROVED** [review]

**Pull request merged**